zboss/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+include(${NRFXLIB_DIR}/common.cmake)
+
+# Common includes, not related to ZBOSS source code.
+zephyr_include_directories(src/zb_error)
+zephyr_include_directories(include/addons)
+
+# Source files
+zephyr_sources(src/zb_error/zb_error_to_string.c)
+
+# Suppress bitfield compatibility warning
+zephyr_compile_options(-Wno-packed-bitfield-compat)
+
+# Add compile-time definition, indicating which ZBOSS library will be linked
+if (CONFIG_ZIGBEE_ROLE_END_DEVICE)
+  zephyr_compile_definitions(ZB_ED_ROLE)
+endif()
+
+
+if ((NOT DEFINED ZEPHYR_PLATFORM_NCS_MODULE_DIR) OR
+    (CONFIG_ZBOSS_LIBRARY_PRECOMPILED))
+
+  zephyr_include_directories(include)
+  zephyr_include_directories(include/zcl)
+  zephyr_include_directories(include/ha)
+  zephyr_include_directories(include/se)
+  zephyr_include_directories(include/osif)
```

Comment:
This should be in dedicated zephyr library.

As this is also needed in `subsys/zigbee/` because of include directories, a Zephyr interface library can be used to linking to.
For inspiration, look here:
https://github.com/NordicPlayground/fw-nrfconnect-zephyr/blob/master/lib/posix/CMakeLists.txt
https://github.com/zephyrproject-rtos/mbedtls/blob/master/CMakeLists.txt

Reply:
Thank you for the set of inspiring links!